driver-core/src/test/java/com/datastax/driver/core/policies/DefaultRetryPolicyIntegrationTest.java

```diff
@@ -146,4 +149,83 @@ public void should_rethrow_unavailable_in_no_host_available_exception() {
             whiteListedCluster.close();
         }
     }
+
+    @Test(groups = "short")
+    public void should_try_next_host_on_client_timeouts() {
+        cluster.getConfiguration().getSocketOptions().setReadTimeoutMillis(1);
+        try {
+            scassandras
+                .node(1).primingClient().prime(PrimingRequest.queryBuilder()
+                    .withQuery("mock query")
+                    .withThen(then().withFixedDelay(1000L).withRows(row("result", "result1")))
+                    .build());
+            scassandras
+                .node(2).primingClient().prime(PrimingRequest.queryBuilder()
+                    .withQuery("mock query")
+                    .withThen(then().withFixedDelay(1000L).withRows(row("result", "result2")))
+                    .build());
+            scassandras
+                .node(3).primingClient().prime(PrimingRequest.queryBuilder()
+                    .withQuery("mock query")
+                    .withThen(then().withFixedDelay(1000L).withRows(row("result", "result3")))
+                    .build());
+            try {
+                query();
+                fail("expected a NoHostAvailableException");
+            } catch (NoHostAvailableException e) {
+                assertThat(e.getErrors().keySet()).hasSize(3).containsExactly(
```

Comment:
The ordering of keySet() is not going to be determinate since it depends on internal map implementation which may vary from JDK implementation,  containsOnly should be adequate here (I think?) so i'll change it.